### DISCUSSION TITLE
libsndfile and mpeg support

### DISCUSSION THREAD

**Discussion by dyfer** [Homebrew/discussions#3323] [Everyday usage]
Hello,
Starting with version 1.1.0 `libsndfile` includes support for mpeg formats. The support for mpeg requires additional dependencies (`lame` and `mpg123`). As far as I can tell, libsndfile from homebrew is not built with mpeg support. Would it be a welcome addition to include these dependencies and in turn enable mpeg support in libsndfile? I can't find any prior conversations on this topic so I wanted to get some feedback before making a PR.

**Comment by SMillerDev** [comment]
Is it enabled by default from upstream?

**Reply by dyfer** [reply]
IIUC, mpeg support is [not disabled](https://github.com/libsndfile/libsndfile/blob/master/configure.ac#L159-L160) by default and it will get built if the dependencies are available. It seems that the support for mpeg gets [turned off automatically](https://github.com/libsndfile/libsndfile/blob/master/configure.ac#L415-L443) when dependencies are not available.

To test this, I tried to install the dependencies and then build libsndfile from source, but it seems that it doesn't find the mpg123 library:

<details>

lame and mpg123 are installed:
```
% brew info lame
lame: stable 3.100 (bottled)
High quality MPEG Audio Layer III (MP3) encoder
https://lame.sourceforge.io/
/opt/homebrew/Cellar/lame/3.100 (27 files, 2.2MB) *
  Poured from bottle on 2021-03-19 at 14:06:12
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/lame.rb
License: LGPL-2.0-or-later
==> Analytics
install: 41,059 (30 days), 125,637 (90 days), 495,154 (365 days)
install-on-request: 488 (30 days), 1,624 (90 days), 7,338 (365 days)
build-error: 0 (30 days)

% brew info mpg123
mpg123: stable 1.29.3 (bottled)
MP3 player for Linux and UNIX
https://www.mpg123.de/
/opt/homebrew/Cellar/mpg123/1.29.3 (33 files, 1.9MB) *
  Poured from bottle on 2022-06-01 at 10:40:55
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/mpg123.rb
License: LGPL-2.1-only
==> Analytics
install: 2,146 (30 days), 6,885 (90 days), 41,256 (365 days)
install-on-request: 367 (30 days), 1,255 (90 days), 11,719 (365 days)
build-error: 4 (30 days)
```
but mpg123 is not found (lame is found though):
```
% brew reinstall --verbose --build-from-source libsndfile
...
checking for flac >= 1.3.1 ... yes
checking for ogg >= 1.3.0 ... yes
checking for vorbis >= 1.2.3 ... yes
checking for vorbisenc >= 1.2.3 ... yes
checking for opus >= 1.1 ... yes
checking for lame/lame.h... yes
checking for library containing lame_set_VBR_q... -lmp3lame
checking for libmpg123 >= 1.25.10 ... no
configure: WARNING: "MPEG support selected but external MPG123 library cannot be found.

configure: WARNING: *** MPEG support disabled.
```

</details>

I'm not sure why libmpg123 is not found. It does show up in `pkg-config`.

**Reply by SMillerDev** [reply]
If it's enabled if available it might be worth it to add the dependency, but that would be better discussed in a pull request.

**Reply by dyfer** [reply]
Hello again, I created a PR for this: https://github.com/Homebrew/homebrew-core/pull/105703
